### DISCUSSION TITLE
large cohorts and multi batches

### DISCUSSION THREAD

**Discussion by NicoDrou** [vdemichev/DiaNN#616] [Q&A]
Dear Vadim,

First of all thank you for having developed such an amazing tool. And thank you to take time to answer our questions!

I come from the metabolomics field I have recently started in Proteomics, so please accept my apologies for my perhaps trivial questions.

I would have a few question about processing multiple batches.
From the tutorial and the forum, I understand we should restrict the fragments used for the quantification using the command lines --restrict-fr and --no-fr-selection. If I understand correctly, the selection of the fragment will switch from a selection based on quality assessment to an exclusion list from the ExcludeFromAssay spectral library column. but it is not clear for me where this ExcludeFromAssay spectral library column comes from and what parameters are considered here.
Also it seems the --restrict-fr command is incompatible with the MBR, but I don't understand why :
WARNING: --restrict-fr is incompatible with MBR and smart/full profiling and will be disabled

And finally a more general question about the cross-run normalization. I completely agree a normalization should be done at the precursor level when dealing with large cohorts before extracting the MaxLFQ and therefore the cross run normalisation should be turned-off. But the mode's names are not very explicit and for my understanding, I would like to know for my understanding, what exactly are the different normalisation modes.

Many thanks in advance for your help and your time.

Best regards,
Nicolas

**Comment by vdemichev** [comment]
Hi Nicolas,

You can always process multiple batches in one go in DIA-NN. The 'incremental' processing described in docs is for a situation when new patient samples keep arriving and reprocessing everything is inpractical.

--gen-fr-restriction can produce ExcludeFromAssay
--restrict-fr does not work with MBR because MBR generates a new library. You can always reproduce the MBR manually with two pipeline steps though (results will not be identical but will be similar in performance).

Cross-run normalisation in general should always be enabled, unless you have some fancy experiment like SEC. But you can always add extra normalisation on top of DIA-NN's. RT-dependent is always the recommended mode, it alleviates problems with sample prep which differentially affect hydrophobic/hydrophilic peptides.

Best,
Vadim

**Reply by NicoDrou** [reply]
Dear Vadim,
Many thanks for your answer.
The incremental processing is exactly the situation I am. 
I am a bit puzzled and I now have even more questions (Sorry for that)

1) How can I reproduce the MBR manually with two pipeline steps? Do you do a first step with MBR enabled and a second patch with MBR disabled using the report-lib of the first step as spectral library? 
2) Do you mind sharing a .pipeline as an example to understand what you mean here? 
3) with such a combined pipeline, what parameters should I used to filter the main report? 

and finally
4) According to the tutorial, the incremental processing is a situation where all DIA-NN normalization should be disabled. Is it still true? 
5) Could you elaborate the mechanism of the RT-dependent normalisation? Does DIA-NN weight differentially the hydrophilic peptides in the identification process in order to have more accurate quantification and more complete data? 

Many thanks again for your help and your time!

Best regards,
Nicolas

**Reply by vdemichev** [reply]
To do sth similar to MBR, need to analyse the experiment with DIA-NN instructed to generate a spectral library, and then reanalyse it with this library.

Please see the docs on filtering. In general, Lib.PG.Q.Value <= 0.01 will work well, nothing in addition to that.

4. In this scenario the data must be further processed in R or Python, including the normalisation. Yes, possible to still leave DIA-NN's normalisation one. However, in 99% cases it's better to just analyse all files in one go instead of doing incremental processing.
5. It detects a situation that hydrophobic peptides, for example, have low quantities in a particular run, and corrects their quantities. 

Best,
Vadim

**Reply by NicoDrou** [reply]
Dear Vladim,
Thank you very much for answering my questions.

I am still not sure to understand the 2 steps-pipeline you suggest.
I guess the second pipeline need to include the command --gen-fr- and --restrict-fr and use the .tsv library created at the first step. 
but for the first step, i should process my raw data with MBR enabled to create a .tsv library.

Is it right?

Many thanks again for your help
Regards,
Nicolas

**Comment by NicoDrou** [comment]
Dear Vladim,
I tried a 2-step pipeline but I have weird results and I am pretty sure I did it wrong.

step1:
![image](https://user-images.githubusercontent.com/124893841/220408228-51bf10f8-c4f1-4cb9-a07a-09edd4afa9f1.png)
Step2: 
![image](https://user-images.githubusercontent.com/124893841/220408431-545d0424-740d-4791-9049-b23495135b0f.png)

1) in the first step, the library already contains the ExcludeFromAssay column even without the --gen-fr-restriction command
2) in the second library and main report of Step 2, the Gene column is missing. This information is present in the step1 library and should be retrieved for the step 2, isn't it?
3) because i don't have the gene information at the second step, I can only compare the protein groups between the 2 steps, and they only present 50% protein group in common. 25% are specific to the first step and 25% are left to the second step. However, many proteins of both protein groups are common. So I guess the protein are grouped differently but I don't understand why. 

Most likely, I didn't understand your thought about the MBR-like using a 2 steps-pipeline. 
Could you give me more details about your suggestion?

**Reply by vdemichev** [reply]
Hi Nicolas, this incremental processing is meant for very specific applications, are you sure it's needed in your case? I will look into settings above.

**Reply by vdemichev** [reply]
In step 1 you can uncheck 'MBR'. Library generation better set to 'IDs, RT and IM'.
1. Yes. --gen-fr-restriction applies to the original library, the one in 'Spectral library' field. Don't need it here.
2. That's because FASTA is not provided. The library created in step 1 contains sequence IDs and gene IDs, but no matching between those, hence once protein grouping is done in step 2, DIA-NN cannot figure out which genes to retain - need FASTA for that.

Best,
Vadim

**Reply by NicoDrou** [reply]
Dear Vadim,

Thank you very much.

Based on what i read so far, I think this is a necessary step. Our goal is not only to be able to compare samples among a cohort, but also we want to be able to compare the samples across cohorts.  
If you don't mind, I would like to exchange more with you about this project as you most likely are ahead of us on this topic. 

Accidentally I already reprocessed the data with no MBR during the first step, a fasta file at the second but smart profiling at both. Therefore, I obtained the same number of proteins and a very close data completeness compared to MBR. 
With your new input, I will try again with the Library generation set to 'IDs, RT and IM'.
Would a classifier mode set at "double pass" (step one, two or both) be beneficial here? 

Many thanks again for your help.
Regards,
Nicolas
